API-GUIDELINE.md

```diff
@@ -0,0 +1,254 @@
+# API GUIDELINES
+
+## Overview
+As commercetools is 100% API First, it's important to have a well designed developer experience.
+Additionally we aim to achieve more automation and improve the review process of new changes in the APIs.
+
+With this in mind, we have created "API Guidelines" that aims to help developers to design APIs:
+
+ - simple
+ - consistent 
+ - self-explanatory
+ 
+The Keywords to have a good Guidelines are:
+ 
+ - Consistency => following the Guidelines
+ - Uniformity => checking in the code if there are other properties with similar name and adjust in base of it or improve it
+ - Explicability => explain with a few words the purpose of the property avoiding abbreviations (there are some [exceptions](#exceptions)).
+
+## Scope
+The purpose of these guidelines is to achieve the definition of consistent practices and patterns for all API endpoints.
+
+## URI
+Our URIs are composed by:
+   
+    Base Uri like:
+        https://{service}.{region}.{cloudProvider}.commercetools.com 
+
+    {projectKey}
+
+    resources
+
+    by ID
+
+    by external ID with equals e.g. customers/key={key}
+
+See our documentation for more details https://docs.commercetools.com/http-api.html#hosts and  https://docs.commercetools.com/http-api-authorization#requesting-an-access-token-using-commercetools-oauth2-server 
+
+## Resources
+Resource is an object or representation of something, which has some associated data with it and can be helped by methods to operate on it.
```

Comment:
Do we want to be that generic or rather be more specific to our business domain?
```suggestion
A Resource is a data collection object or a data representation of an entity in the e-commerce domain, which is specified by data fields and methods to operate on them.
```